### ISSUE TITLE
Building in fresh git clone fails: sha512 checksum failure on SuiteSparse-5d7d1d69b9bc6597dcaee80a4c2e596237523b79.tar.gz

### ISSUE THREAD

**Issue by fingolfin** [JuliaLang/julia#43896]
I am getting this on both a Mac and a Linux machine, with the current git master:
```
$ make
...
  ERROR: sha512 checksum failure on SuiteSparse-5d7d1d69b9bc6597dcaee80a4c2e596237523b79.tar.gz, should be:
      316af7423ac15c78714d2079b3a9959d49cb59f30b5b99b654714e54d0667c29
      bd3f08c1bd8e9b9890bdc22779e71e0e352ecf1a81ad61fb85ca745eea2ea855
  But `sha512sum /Users/horn/Projekte/Julia/julia/stdlib/srccache/SuiteSparse-5d7d1d69b9bc6597dcaee80a4c2e596237523b79.tar.gz | awk '{ print $1; }'` results in:
      1cce0d88be2f28c8b1d49bef65899ae6a4620e0dc18cb509deb1d248d7b0efe0
      d383be22744ace4be349f51d6ab91d7b597eccb2c38dc1690327f59d649f3be4
  This can happen due to bad downloads or network proxies, please check your
  network proxy/firewall settings and delete
  /Users/horn/Projekte/Julia/julia/stdlib/srccache/SuiteSparse-5d7d1d69b9bc6597dcaee80a4c2e596237523b79.tar.gz
  to force a redownload when you are ready
===============================================================================
make[1]: *** [SuiteSparse-5d7d1d69b9bc6597dcaee80a4c2e596237523b79/source-extracted] Error 2
make: *** [julia-stdlib] Error 2
$
```
I tried to force-redownload as advised but no change. The tarball looks fine in general (not corrupt).

**Comment by IanButterworth** [comment]
Same on MacOS

**Comment by IanButterworth** [comment]
@ViralBShah perhaps?

**Comment by cortner** [comment]
I still get the same problem (M1) - should this be fixed now on v1.7.3 tag?

**Comment by MilesCranmer** [comment]
I see this on 1.7.3 and 1.6.7 which means I can't build old versions from source. Here's a minimal dockerfile to reproduce:

```dockerfile
FROM ubuntu:20.04

ENV DEBIAN_FRONTEND=noninteractive

RUN apt-get update && apt-get install -y --no-install-recommends \
    build-essential \
    cmake \
    git \
    wget \
    ca-certificates \
    curl \
    gpg-agent \
    software-properties-common \
    python3 \
    python3-dev \
    tar \
    xz-utils \
    gfortran


RUN wget https://apt.llvm.org/llvm.sh && chmod +x llvm.sh && ./llvm.sh 13

ARG JLVERSION=1.7.3
ARG PROCS=4

WORKDIR /julia
ENV BUILDDIR=/julia

RUN git clone https://github.com/JuliaLang/julia ${BUILDDIR} && \
    cd ${BUILDDIR} && \
    git checkout v${JLVERSION}

# Build:
RUN make -j ${PROCS}
```

Run with e.g., `docker build --build-arg JLVERSION=1.7.3 --build-arg PROCS=8 -t julia .` The error is:

```
===============================================================================
  ERROR: sha512 checksum failure on Statistics-5256d570d0a554780ed80949c79116f47eac6382.tar.gz, should be:
      15b2ed0b269605b1dab150fccc8e202829278bc4920293b711c611627cd4a617
  But `sha512sum /julia/stdlib/srccache/Statistics-5256d570d0a554780ed80949c79116f47eac6382.tar.gz | awk '{ print $1; }'` results in:
      67373895acd1d45a0b37557e8c0cd123673d75944cd08216cffa81dc5186c0fc
      f4de867d2661588eaf14d15f9c26a83b31ccb184c1a00f1b21fc792959d5f90e
  This can happen due to bad downloads or network proxies, please check your
  network proxy/firewall settings and delete
  /julia/stdlib/srccache/Statistics-5256d570d0a554780ed80949c79116f47eac6382.tar.gz
  to force a redownload when you are ready
===============================================================================
      cbbab6ff5ac0e654aac722f56d9b81768cfffd1d84a6ece2360f32d70650734a
make[1]: *** [Makefile:49: Statistics-5256d570d0a554780ed80949c79116f47eac6382/source-extracted] Error 2
make[1]: *** Waiting for unfinished jobs....
100   477  100   477    0     0    766      0 --:--:-- --:--:-- --:--:--   766
100  412k    0  412k    0     0  1763k      0 --:--:-- --:--:-- --:--:-- 1763k
100  221k  100  221k    0     0   281k      0 --:--:-- --:--:-- --:--:-- 14.3M
make: *** [Makefile:64: julia-stdlib] Error 2
make: *** Waiting for unfinished jobs....
```
I see the same issue on my institute cluster as on my laptop behind my home wifi, so I don't think it's a proxy error.
